content/realtime/channels.textile

```diff
@@ -919,6 +997,7 @@ bq(definition#publish-data).
   java:     void publish(String name, Object data, "CompletionListener":#completion-listener listener)
   csharp:   void Publish(string name, object data, Action<bool, ErrorInfo> callback = null, string clientId = null)
   objc,swift: publish(name: String?, data: AnyObject?, callback: (("ARTErrorInfo":/realtime/types#error-info?) -> Void)?)
+  flutter:  Future<void> publish(name: String name, data: Object data, message: "Message":#message message)
```

Comment:
Is this correct?  It looks odd with the arg duplicated?  I may be wrong as I am no Dart expert, but wouldn't:

```dart
Future<void> publish(name: String, data: Object, message: "Message":#message)
```

.. be better?

Reply:
@mattheworiordan do I need to expose the function signature here?

In that case it will have to be, as all the arguments passed here are named (like kwargs in python)

```dart
Future<void> publish({
  name: String,
  data: Object,
  message: "Message":#message
  messages: List<"Message":#message>
});
```

Reply:
I agree with @kavalerov's [comment](https://github.com/ably/docs/pull/989#pullrequestreview-535400038) that this doesn't fall within the scope of this PR. Let's capture this concern as an issue in [ably-flutter](https://github.com/ably/ably-flutter), please, @tiholic - if any API changes are made then we'll have to make a subsequent docs change.

Reply:
@kavalerov @QuintinWillison our current implementation in SDK uses named params, but I wasn't sure on what to represent in docs. Should we be showcasing method signature here? Or its usage?

Reply:
@tiholic, I think "usage", matching the surrounding examples from other client libraries.

Reply:
@Quintin
In that case we're good to resolve this conversation, I guess _( unless the syntax isn't clear enough! )_